### PR TITLE
Minor Geometry optimization

### PR DESCRIPTION
**Pull request by Dr15Jones**
# This Pull request:
- Returns the newly added node from call to TGeoVolume::AddNode
- Allows an option to avoid voxelation

## Changes or fixes:

Profiling geometry building in CMSSW turned up two issues
1. Doing a linear lookup for newly added nodes to a TGeoVolume was very slow2. 
2. CMS uses DD4Hep (which uses ROOT) as a geometry description system not as a geometry navigation system. The building of the voxels was slow and is never used in CMSSW. 

This change was temporarily incorporated into CMS' development branch and did not cause any noticeable problems and did give a speed boost.

## Checklist:

- [ x] tested changes locally
- [ ] updated the docs (if necessary)

This PR fixes #

### CODE REVIEW

**Comment by Dr15Jones** [comment]
@smuzaffar FYI

File: geom/geom/src/TGeoManager.cxx [hunk 1]
```diff
@@ -1434,7 +1434,15 @@ void TGeoManager::CloseGeometry(Option_t *option)
    if (fNLevel<30) fNLevel = 100;
 
 //   BuildIdArray();
-   Voxelize("ALL");
+   if (!opt.Contains("v")) {
```
Comment (agheata) [inline geom/geom/src/TGeoManager.cxx:1437]: The option name is a bit counter-intuitive as one would expect "v" to voxelize and not to disable it (notably used by default in the Export method), maybe use "nv". I would add a comment line explaining it in the code just before its use.

**Review by agheata — CHANGES_REQUESTED** [review]
Thanks Chris, I think this is useful, I've made a few comments in line with the code.

File: geom/geom/src/TGeoManager.cxx (hunk 1 above)
Comment (Dr15Jones) [reply geom/geom/src/TGeoManager.cxx]: changed

File: geom/geom/src/TGeoManager.cxx [hunk 2]
```diff
@@ -1434,7 +1434,15 @@ void TGeoManager::CloseGeometry(Option_t *option)
...
+      Voxelize("ALL");
+   } else {
+      TGeoVolume *vol;
+      TIter next(fVolumes);
+      while ((vol = (TGeoVolume *)next())) {
+         vol->SortNodes();
+      }
```
[the remark this answers is not in the document]
Comment (Dr15Jones) [reply geom/geom/src/TGeoManager.cxx:1447]: added to reading back

**Comment by smuzaffar** [comment]
@Dr15Jones @agheata , can you please backport it to root v6.26 and v6.24?

**Comment by agheata** [comment]
> @Dr15Jones @agheata , can you please backport it to root v6.26 and v6.24?

OK, done, thanks @Dr15Jones

**Pull request merged**